### PR TITLE
Get top ranking in frontend

### LINKED ISSUE #38: Users should be able to get ranking

### PR DESCRIPTION
**Pull request by EricLung**
Closes #38.

### CODE REVIEW

File: front-end/src/store/users.js [hunk 1]
```diff
@@ -87,6 +100,38 @@ export default {
...
+      state.listRank.loading = true
+
+      // testing
+      // state.listRank.users.push({
+      //   Rank: 1,
+      //   Name: 'root',
+      //   Score: 100,
+      //   Last_solve: 1
+      // })
+      // state.listRank.loading = false
```
Comment (samueltangz) [inline front-end/src/store/users.js]:
```
state.listRank.users.push({
  rank: 1,
  name: 'root',
  score: 100,
  lastSolve: new Date() // this should be a time
})
state.listRank.loading = false
```

File: front-end/src/store/users.js [hunk 2]
```diff
@@ -87,6 +100,38 @@ export default {
...
+
+      try {
+        const res = await axios.get('/api/users/ranking', {
+          params: {
+            pageNum: page
```
Comment (samueltangz) [inline front-end/src/store/users.js]: use `page` instead of `pageNum`?

File: front-end/src/views/Ranking.vue [hunk 3]
```diff
@@ -31,27 +40,25 @@
...
     return {
       rankingTableFields: [
-        'name', 'score'
-      ],
-      users: [
-        {
-          id: 1,
-          name: 'root'
-        }
+        'Rank', 'Name', 'Score', 'Last_solve'
```
Comment (samueltangz) [inline front-end/src/views/Ranking.vue]: `'rank', 'name', 'score', 'lastSolve'`

File: front-end/src/views/Ranking.vue [hunk 4]
```diff
@@ -60,10 +67,14 @@ export default {
   },
   watch: {},
 
-  created () {},
+  created () {
+    store.dispatch('users/listRank', 1)
```
Comment (samueltangz) [inline front-end/src/views/Ranking.vue]: will it support multiple pages?

Comment (EricLung) [reply front-end/src/views/Ranking.vue]: multiple pages will implement later

**Pull request merged**